src/Core/Grid/Data/Factory/DoctrineGridDataFactory.php

```diff
@@ -93,9 +96,37 @@ public function getData(SearchCriteriaInterface $searchCriteria)
             'search_criteria' => $searchCriteria,
         ]);
 
-        $records = $searchQueryBuilder->execute()->fetchAll();
         $recordsTotal = (int) $countQueryBuilder->execute()->fetch(PDO::FETCH_COLUMN);
+        $searchOffset = (int) $searchCriteria->getOffset();
+
+        // If offset is out of range, we compute and apply a valid offset.
```

Comment:
The code is legit, however this should be done before creating the query builder, since the limit,offset parameters are applied during `getSearchQueryBuilder` call